CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+
+- Testing
+  - [ ] The PR has sufficient tests and test cases, with new code meeting at least 70% coverage
+  - Are new methods, properties, parameters or return types added, removed, or changed?
+    - [ ] The component has new tests, test cases, or modified tests that reflect the changes
+  - Does PR include a component from another library?
+    - [ ] The developer added test cases to cover integrating with library
+    - [ ] If extending a third-party component, you have validated that is the best design solution
+- HTML/CSS/UI Controls
+  - [ ] HTML markup and CSS classes are semantic and meaningful
+  - [ ] Styles are applied by classes, not through style tags on elements
+  - [ ] CSS class names reflect structure, state, or intent rather than specific styles
+  - [ ] Markup choices reflect intent, e.g., `button` for user actions, `a` for navigation
+  - Is the developer creating a new UI component or extending an existing component?
+    - [ ] There is no existing UI component that can be used or modified
+    - [ ] Keyboard interactions follow the conventions listed in the WAI-ARIA Authoring Practices
+    - [ ] Aria roles and attributes are applied as necessary
+  - Code Smells
+    - [ ] Lots of custom CSS classes require careful review
+    - [ ] Component inheritance requires careful review
+    - [ ] Class names with specific styles require careful review
+    - [ ] Large numbers of divs or nested divs require careful review
+    - [ ] No aria tags on custom controls with many event handlers requires careful review
+- Modularity & Decomposition
+  - Is the PR for a new feature?
+    - [ ] The feature is written as a standalone module
+    - [ ] The feature is lazy loaded unless required during App initialization. This applies to both top level routing and children of feature modules
+    - [ ] Only the code necessary to execute the feature is included in the module
+    - [ ] Features do not expose internal implementation via exports
+    - [ ] Modules consuming shared components explicitly import their dependencies (Do not assume deps are provided by a higher module like App)
+  - Is the PR for a UI or data-access component, e.g., a class in app-components or api-kit?
+    - [ ] The component is declared and exported in its own module
+    - [ ] The component is not included in a "big bang" module (AppComponentsModule, ApiKitModule). Each consuming module must explicitly import all dependencies.
+  - Is the code a utility library?
+    - [ ] Any Angular utilities follow the same module rules as UI and data-access components
+    - [ ] JS/TS POJOs or functions should be exported from their respective files inside an appropriately named folder that includes unit tests, interfaces, or any other code related to that.
+  - [ ] Barrel files (index.js/ts) are avoided (this creates complications with Angular Package Format)
+  - Code Smells
+    - [ ] Classes and templates that exceed ~1-300 LOC require careful review
```

Comment:
@thomastighe How about

"Classes and templates that exceed a few hundred lines of code require careful review"

Reply:
@thomastighe We aren't going to enforce the number on developers, this is a checklist for reviewers. If you see a file that exceeds a few hundred lines of code, the developer should be able to provide a reasonable justification for that. Otherwise, there's probably a lot of duplication and tight coupling that should be pulled apart before the code is merged.

Reply:
I understand that this is a guide but I just thought that looked odd. We should encourage the breaking down into smaller components.